pkg/collector/python/init.go

```diff
@@ -203,20 +205,27 @@ func Initialize(paths ...string) error {
 		return fmt.Errorf("%s", err)
 	}
 
-	// store the Python version after killing \n chars within the string
-	if res := C.get_py_version(six); res != nil {
-		PythonVersion = strings.Replace(C.GoString(res), "\n", "", -1)
+	// Lock the GIL
+	glock := newStickyLock()
+	pyInfo := C.get_py_info(six)
+	glock.unlock()
 
+	// store the Python version after killing \n chars within the string
+	if pyInfo != nil {
+		PythonVersion = strings.Replace(C.GoString(pyInfo.version), "\n", "", -1)
 		// Set python version in the cache
 		key := cache.BuildAgentKey("pythonVersion")
 		cache.Cache.Set(key, PythonVersion, cache.NoExpiration)
+
+		PythonPath = C.GoString(pyInfo.path)
+		C.six_free(six, unsafe.Pointer(pyInfo.path))
```

Comment:
I had to go very deep (down to the implementation of `*Two::getPyInfo()` and the C Py docs on `Py_GetVersion` at https://docs.python.org/2/c-api/init.html#c.Py_GetVersion) to understand why we're not `free`ing `pyInfo.Version` here.

Would it make sense to document this somewhere inline in the code? I'm thinking it could be documented inline in the definition of the `py_info_t` struct, what do you think?

Reply:
`versions` is `const` in `py_info_t`, I can add a comment if you think it's not enough

Reply:
right, thanks for pointing that out, I really need to brush up my knowledge of C 😅